### PR TITLE
refactor!(NODE-3295): Support node 12+ and reconfigure CI

### PR DESCRIPTION
**Pull request by nbbeeken**
## Changes

- engine version reccomends node 12+
- removed node 10 from test CI
- simplified CI matrix
  - windows vs2019
  - ubuntu 18.04
  - macos 10.14
- FLE runs on all systems ~(temporarily disabled pending release of required changes)~
- OCSP/TLS tests run on all OSes
- Windows testing all versions of mongodb, there was a comment about skipping 4.2 but it is working now?
- Extra variants moved to ubuntu for conistency. We can totally change our "default" OS version with a smalll script fix here. Ubuntu is required for LB so making it default makes sense there.
- npm/node missing issue fixed by consolidating nvm-init to a script, it has to be setup differently on windows than other systems. 
- set PROJECT_DIRECTORY on tasks that don't use PREPARE_SHELL
- [refactored backtic usage](https://github.com/koalaman/shellcheck/wiki/SC2006)
- [refactored assignment and export onliners](https://github.com/koalaman/shellcheck/wiki/SC2155)

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -4,10 +4,11 @@
     "checkJs": false,
     "strict": true,
     "alwaysStrict": true,
-    "target": "ES2018",
+    "target": "ES2019",
     "module": "commonJS",
     "moduleResolution": "node",
-    "lib": ["ES2018"],
+    "skipLibCheck": true,
+    "lib": ["es2019", "es2020.promise", "es2020.bigint", "es2020.string"],
```
Comment (nbbeeken) [inline tsconfig.json:11]: Reference: https://github.com/tsconfig/bases/blob/main/bases/node12.json

File: package.json [hunk 2]
```diff
@@ -83,7 +83,7 @@
   },
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=10"
+    "node": ">=12.9.0"
```
Comment (nbbeeken) [inline package.json:86]: This choice is based on:

Node 12.9 updated its v8 version: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md#2019-08-20-version-1290-current-targos

And these are the changes that it brings about: https://v8.dev/blog/v8-release-76

File: .evergreen/init-nvm.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#! /usr/bin/env bash
```
Comment (emadum) [inline .evergreen/init-nvm.sh:1]: Good call extracting this! 👍

**Review by emadum — APPROVED** [review]
LGTM, nice cleanup and future-proofing! 🚀

File: .evergreen/generate_evergreen_tasks.js [hunk 4]
```diff
@@ -3,62 +3,40 @@ const fs = require('fs');
 const yaml = require('js-yaml');
 
 const LATEST_EFFECTIVE_VERSION = '5.0';
-const MONGODB_VERSIONS = ['latest', '4.4', '4.2', '4.0', '3.6', '3.4', '3.2', '3.0', '2.6'];
-const NODE_VERSIONS = ['dubnium', 'erbium', 'fermium'];
+const MONGODB_VERSIONS = ['latest', '5.0', '4.4', '4.2', '4.0', '3.6', '3.4', '3.2', '3.0', '2.6'];
```
[the remark this answers is not in the document]
Comment (nbbeeken) [reply .evergreen/generate_evergreen_tasks.js:6]: Something like commit messge?
```
refactor!(NODE-3295): Support node 12+ and update CI targets

[add some notes about each point hit here, tsconfig, tests, os versions]
Resolves NODE-3306
```
and I can add a link on that ticket

File: .evergreen/generate_evergreen_tasks.js [hunk 5]
```diff
@@ -3,62 +3,40 @@ const fs = require('fs');
...
+const LOWEST_LTS = 'erbium';
+const NODE_VERSIONS = [LOWEST_LTS, 'fermium'];
```
[the remark this answers is not in the document]
Comment (nbbeeken) [reply .evergreen/generate_evergreen_tasks.js]: Good point, done!

**Review by durran — APPROVED** [review]

**Pull request merged**